### PR TITLE
make ok() error informative when fauxpas isn't installed

### PR DESCRIPTION
**Pull request by maelle**
cf #108

I'm not sure how to add a test.
* `requireNamespace` is a base function so cannot be mocked with `testthat::with_mock()` cf https://www.tidyverse.org/articles/2017/12/testthat-2-0-0/ so I'd need to add a dependency on `mockery` or `mockr`
* maybe I could make a `check_package()` function [like what `desc` has](https://github.com/r-lib/desc/blob/42b9578f374bf685610b1efb05315927ae236d5b/R/utils.R#L85) just so I can mock it with `with_mock`?

### CODE REVIEW

**Comment by maelle** [comment]
Or maybe you could just list `fauxpas` under Imports: :-)

**Comment by sckott** [comment]
thanks! 

i do want to keep `fauxpas` in Suggests just to reduce imports here. I think we should take approach of the `raise_for_status` method in `HttpResponse`   https://github.com/ropensci/crul/blob/master/R/response.R#L167-L172  Sound good?

**Comment by maelle** [comment]
So what should be the replacement for the `fauxpas` code when it is not installed? I am not sure I follow. :-)

**Comment by sckott** [comment]
it would be like this:

```r
if (!requireNamespace("fauxpas", quietly = TRUE)) {
  x <- httpcode::http_code(code = self$status_code)
  stop(sprintf("%s (HTTP %s)", x$message, x$status_code), call. = FALSE)
} else {
  fauxpas::http(self, behavior = "stop")
}
```

`httpcode` is in Imports, so if `fauxpas` is not installed, we use `httpcode`

**Comment by maelle** [comment]
I've played around a bit but it turns out I'm not able to make this work, I'm sorry.

**Comment by maelle** [comment]
Ok, I think now I understand, I was indeed missing something obvious :woman_facepalming: 

How should this be tested?

File: R/ok.R [hunk 1]
```diff
@@ -50,8 +50,15 @@ ok.character <- function(x, status = 200L, info = TRUE, ...) {
 ok.HttpClient <- function(x, status = 200L, info = TRUE, ...) {
   assert(info, "logical")
   assert(status, "integer")
-  find_status <- tryCatch(fauxpas::find_error_class(status), 
-    error = function(e) e)
+  
+  if (!requireNamespace("fauxpas", quietly = TRUE)) {
```
Comment (sckott) [inline R/ok.R]: woops, sorry, I meant to just use httpcode, so can get rid of the if/else and just use httpcode, sound good?

Comment (sckott) [reply R/ok.R]: this makes testing easier as well

Comment (maelle) [reply R/ok.R]: Aaah it makes sense! Done!

Comment (sckott) [reply R/ok.R]: thanks

**Comment by sckott** [comment]
can test i think like 

```r
ok(hb("/status/404"), status = 567L)
```

**Comment by sckott** [comment]
thanks so much @maelle 🚀

**Pull request merged**